src/components/pagination/pagination.jsx

```diff
@@ -0,0 +1,172 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import injectSheet from 'react-jss';
+import { Icon } from '../../';
+import Page from './page';
+import Stepper from './stepper';
+import Ellipsis from './ellipsis';
+import styles from './pagination-styles';
+
+const getRandomString = () =>
+  Math.random()
+    .toString(36)
+    .substr(2, 9);
+
+class Pagination extends Component {
+  constructor(props) {
+    super(props);
+    const { total, limit, selected, anchors, selectedSiblings } = this.props;
+
+    this.state = {
+      selected: parseInt(selected, 10),
+    };
+
+    this.pagesCount = Math.ceil(total / limit);
+    this.pagesToDraw = 1 + anchors * 2 + selectedSiblings * 2; // 1 = the selected item
+    this.advancedPagination = this.pagesCount > this.pagesToDraw;
```

Comment:
Are these (pagesCount, pagesToDraw, advancedPagination) directly or indirectly being used in the render cycle? Then they should probably be in the state instead. 

Also, since they currently rely on props, they should probably be re-calculated if/when props change.